src/CandidateComparator.php

```diff
@@ -5,13 +5,28 @@
 
 class CandidateComparator
 {
+    private $ballot;
     private $candidateIdToRank;
+
+    /**
+     * @param a Ballot. The CandidateComparator will store a copy of the Ballot. The caller of this constructor may
+     * subsequently modify the parameterized Ballot without affecting this CandidateComparator.
+     */
     public function __construct(Ballot $ballot)
     {
+        $this->ballot = clone $ballot;
         $this->candidateIdToRank = $this->makeCandidateIdToRankMap($ballot);
     }
 
     /**
+     * @return a copy of the Ballot that this instance uses to inform its comparisons. The returned Ballot may be
+     * modified without affecting this CandidateComparator.
+     */
+    public function getBallot() : Ballot
+    {
+        return clone $this->ballot;
+    }
+    /**
```

Comment:
Pretty sure the sign convention in this class' compare() method is backwards too.

Reply:
Fixed.